backend/audit/cross_validation/check_finding_prior_references.py

```diff
@@ -0,0 +1,136 @@
+from .errors import (
+    err_prior_no_report,
+    err_bad_repeat_prior_reference,
+    err_prior_ref_not_found,
+)
+from dissemination.models import (
+    Finding,
+    General,
+)
+
+from datetime import date
+
+
+def check_finding_prior_references(sac_dict, *_args, **_kwargs):
+    """
+    Check that prior references numbers point to findings that actually exist
+    in a previously submitted report
+    """
+    # Importing here to avoid circular import
+    from audit.models import SacValidationWaiver
+
+    if SacValidationWaiver.TYPES.PRIOR_REFERENCES in sac_dict.get("waiver_types", []):
+        return []
+
+    all_sections = sac_dict.get("sf_sac_sections")
+    findings_uniform_guidance_section = (
+        all_sections.get("findings_uniform_guidance") or {}
+    )
+    findings_uniform_guidance = findings_uniform_guidance_section.get(
+        "findings_uniform_guidance_entries", []
+    )
+    all_prior_refs = _get_prior_refs(findings_uniform_guidance)
+
+    # No prior reference numbers to validate
+    if not all_prior_refs:
+        return []
+
+    general_information = all_sections.get("general_information")
+    auditee_uei = general_information["auditee_uei"]
+    previous_year = (
+        date.fromisoformat(general_information["auditee_fiscal_period_start"]).year - 1
+    )
+
+    # Prior reference numbers usually refer to findings from the previous
+    # year's report
+    try:
+        previous_year_report = General.objects.get(
```

Comment:
This assumes that the combination of (audit_year, auditee_uei) corresponds to a single record, which is not always the case. For example, `ZZURKVEBSJL1` maps to two reports in 2022 and to two reports in 2023. There are cases where the same auditee_uei maps to multiple reports within a given audit year

Reply:
![image](https://github.com/user-attachments/assets/383b96b5-3637-4676-8a63-5261aaf34a97)

Reply:
Any objections to searching for the prior ref in all of those reports, then?

Reply:
Fixed